instrumentation/mongodb/README.md

```diff
@@ -0,0 +1,35 @@
+# brave-instrumentation-mongodb
+
+This includes [`TraceMongoCommandListener`](src/main/java/brave/mongodb/TraceMongoCommandListener.java), a
+[`CommandListener`](https://mongodb.github.io/mongo-java-driver/3.12/driver/reference/monitoring/#command-monitoring)
+for the Mongo Java driver that will report via Brave how long each command takes, along with relevant tags like the
+collection/view name, the command's name (`insert`, `update`, `find`, etc.).
+
+This instrumentation can only be used with the synchronous MongoDB driver. Do not use it with the asynchronous or
+reactive drivers as tracing data will be incorrect.
+
+## Span properties:
+- `name`: command name (and collection/view name, if available). Examples: `find myCollection`, `listCollections`, etc.
+- `kind`: `CLIENT`
+- `remoteServiceName`: `mongodb-${databaseName}`. Example: `mongodb-myDatabase`.
+- `remoteIpAndPort`: the IP address and port number of the MongoDB server that the command was issued to
+- `error`: `Throwable` in case of failed command
+- Tags:
+  - `mongodb.command.name`: the name of the MongoDB command. Examples: `find`, `listCollections`, etc.
+  - `mongodb.command`: a possibly truncated, JSON version of the full command. Example:
```

Comment:
I understand a lot more now, reading all of things here. I suggest we typically don't tag large things at all, which is why this has come up a bit. For example, in gRPC we added a hook so that users could do this if they want, but instead rely on logging (with trace/spanid correlation) for large data such as payloads. Remember there are other tools like haystack blobs which will also send the request/response body.

Despite us chatting a bit about how we would store this, I think we skipped why, and especially why by default store this at all. I'd like to hear from existing sites about this and if anything make it optional, if not a custom addition if the majority aren't interested in doing it like this (vs logging which is typically built for larger data and could have the whole thing)